### PR TITLE
[Merged by Bors] - feat: `IsCoprime.wronskian_eq_zero_iff`

### PR DESCRIPTION
**Pull request by seewoo5**
Add `IsCoprime.wronskian_eq_zero_iff`: For two polynomials $a, b \in k[x]$ over a field, $W(a, b) = 0$ if and only if $a' = b' = 0$. This is a part of PR #15706.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Polynomial/Wronskian.lean [hunk 1]
```diff
@@ -113,4 +114,40 @@ theorem natDegree_wronskian_lt_add {a b : R[X]} (hw : wronskian a b ≠ 0) :
...
+  constructor
+  · intro h
+    by_cases a_nz : a = 0
+    · rw [a_nz]; simp only [derivative_zero]
+    by_contra deriv_nz
+    have deriv_lt := degree_derivative_lt a_nz
+    have le_deriv := Polynomial.degree_le_of_dvd h deriv_nz
+    have lt_self := le_deriv.trans_lt deriv_lt
+    simp only [lt_self_iff_false] at lt_self
+  intro h; rw [h]; simp
```
Comment (YaelDillies) [inline Mathlib/RingTheory/Polynomial/Wronskian.lean:130]:
```suggestion
theorem dvd_derivative_iff {a : k[X]} : a ∣ derivative a ↔ derivative a = 0 where
  mp h := by
    by_cases a_nz : a = 0
    · rw [a_nz]; simp only [derivative_zero]
    by_contra deriv_nz
    have deriv_lt := degree_derivative_lt a_nz
    have le_deriv := Polynomial.degree_le_of_dvd h deriv_nz
    have lt_self := le_deriv.trans_lt deriv_lt
    simp only [lt_self_iff_false] at lt_self
  mpr h := simp [h]
```

File: Mathlib/RingTheory/Polynomial/Wronskian.lean [hunk 2]
```diff
@@ -113,4 +114,40 @@ theorem natDegree_wronskian_lt_add {a b : R[X]} (hw : wronskian a b ≠ 0) :
...
+    · rw [← dvd_derivative_iff]
+      apply hc.dvd_of_dvd_mul_right
+      rw [← hw]; exact dvd_mul_right _ _
+    · rw [← dvd_derivative_iff]
+      apply hc.symm.dvd_of_dvd_mul_left
+      rw [hw]; exact dvd_mul_left _ _
+  intro hdab
+  cases' hdab with hda hdb
+  rw [wronskian]
+  rw [hda, hdb]; simp only [MulZeroClass.mul_zero, MulZeroClass.zero_mul, sub_self]
```
Comment (YaelDillies) [inline Mathlib/RingTheory/Polynomial/Wronskian.lean:151]: Same thing here ;)

File: Mathlib/RingTheory/Polynomial/Wronskian.lean (hunk 1 above)
Comment (seewoo5) [reply Mathlib/RingTheory/Polynomial/Wronskian.lean]: Thanks, I just learned `mp` and `mpr` ;D

File: Mathlib/RingTheory/Polynomial/Wronskian.lean [hunk 3]
```diff
@@ -113,4 +114,38 @@ theorem natDegree_wronskian_lt_add {a b : R[X]} (hw : wronskian a b ≠ 0) :
   · exact Polynomial.degree_eq_natDegree ha
   · exact Polynomial.degree_eq_natDegree hb
 
+variable {k : Type*} [Field k]
+
+@[simp]
+theorem dvd_derivative_iff {a : k[X]} : a ∣ derivative a ↔ derivative a = 0 where
+  mp h := by
+    by_cases a_nz : a = 0
+    · rw [a_nz]; simp only [derivative_zero]
```
Comment (YaelDillies) [inline Mathlib/RingTheory/Polynomial/Wronskian.lean:123]:
```suggestion
    · simp only [a_nz, derivative_zero]
```

**Review by YaelDillies — APPROVED** [review]
Thanks!

maintainer merge

**Comment by riccardobrasca** [comment]
There are a couple of unaddressed comments, but otherwise LGTM, thanks!

bors d+

**Pull request closed without merging**